modules/test-runner/src/channelProvider/channelProvider.test.ts

```diff
@@ -0,0 +1,106 @@
+  let clientA1: IConnextClient;
+  let tokenAddress: string;
+  let nodeFreeBalanceAddress: string;
+  let nodePublicIdentifier: string;
+  let channelProvider: IChannelProvider;
+
+  beforeEach(async () => {
+    clientA = await createClient();
+    tokenAddress = clientA.config.contractAddresses.Token;
+    nodePublicIdentifier = clientA.config.nodePublicIdentifier;
+    nodeFreeBalanceAddress = xkeyKthAddress(nodePublicIdentifier);
+    channelProvider = await createChannelProvider(clientA);
+    clientA1 = await createRemoteClient(channelProvider);
+  }, 90_000);
+
+  // tslint:disable-next-line:max-line-length
+  test("Happy case: client A1 can be instantiated with a channelProvider generated from client A", async () => {
+    // tslint:disable-next-line:variable-name
+    const _tokenAddress = clientA1.config.contractAddresses.Token;
+    // tslint:disable-next-line:variable-name
+    const _nodePublicIdentifier = clientA1.config.nodePublicIdentifier;
+    // tslint:disable-next-line:variable-name
+    const _nodeFreeBalanceAddress = xkeyKthAddress(nodePublicIdentifier);
+
+    expect(_tokenAddress).toBe(tokenAddress);
+    expect(_nodePublicIdentifier).toBe(nodePublicIdentifier);
+    expect(_nodeFreeBalanceAddress).toBe(nodeFreeBalanceAddress);
+  });
+
+  // tslint:disable-next-line:max-line-length
+  test("Happy case: Bot A1 can call the full deposit → swap → transfer → withdraw flow on Bot A", async () => {
+    const input = { amount: ETH_AMOUNT_SM, assetId: AddressZero };
+    const output = { amount: TOKEN_AMOUNT, assetId: tokenAddress };
+
+    ////////////////////////////////////////
+    // DEPOSIT FLOW
+    await clientA.deposit({ amount: input.amount.toString(), assetId: input.assetId });
+    await clientA.requestCollateral(output.assetId);
+
+    ////////////////////////////////////////
+    // SWAP FLOW
+    const { freeBalanceClientEth, freeBalanceNodeEth } = await swapAsset(
+      clientA,
+      input,
+      output,
+      nodeFreeBalanceAddress,
+    );
+
+    ////////////////////////////////////////
+    // TRANSFER FLOW
+    const transfer = { amount: bigNumberify(ONE), assetId: AddressZero };
+    const clientB = await createClient();
+    await clientB.requestCollateral(AddressZero);
+
+    await asyncTransferAsset(
+      clientA1,
+      clientB,
+      transfer.amount,
+      transfer.assetId,
+      nodeFreeBalanceAddress,
+      {
+        freeBalanceClientA: freeBalanceClientEth,
+        freeBalanceNodeA: freeBalanceNodeEth,
+      },
+    );
+
+    ////////////////////////////////////////
+    // WITHDRAW FLOW
+    const withdraw = { amount: bigNumberify(ONE), assetId: AddressZero };
+    await withdrawFromChannel(clientA1, withdraw.amount.toString(), withdraw.assetId);
+  });
+
+  // tslint:disable-next-line:max-line-length
+  test("Bot A1 tries to call a function when Bot A is offline", async () => {
+    // close channelProvider connection
+    clientA1.channelProvider.close();
```

Comment:
ahhhhh so the `open` / `close` is to simulate going "offline"?

Reply:
yes!